### DISCUSSION TITLE
formatting target variable

### DISCUSSION THREAD

**Discussion by sbpatel2009** [sebp/scikit-survival#245] [Q&A] [answered]
Never mind.  I just had to change dtype.  

Hello all, I want to convert my event - time data into a form suitable for scikit-survival.  This is what I've tried.  Unfortunately, it doesn't work. 
```
event = [True, True, False, True, False]
time = [5, 1, 8, 2, 3]
y = list(zip(event, time))
y = np.array(y, dtype = "O")
```

> array([
>        [True, 5],
>        [True, 1],
>        [False, 8],
>        [True, 2],
>        [False, 3]], dtype=object)
> 

However, the required format is below.  How do I get my data in this form below?

> array([
>           ( True,  5), 
>           ( True, 1), 
>           ( False, 8), 
>           ( True, 2),
>           ( False, 3), dtype=[('Status', '?'), ('Survival_in_days', '<f8')])
> 

This is the error I get when I try to run a model:

```
from sksurv.linear_model import CoxPHSurvivalAnalysis
estimator = CoxPHSurvivalAnalysis()
estimator.fit(df, y))
```

> ValueError: y must be a structured array with the first field being a binary class event indicator and the second field the time of the event/censoring

**Comment by sebp** [comment, marked as answer]
Use sksurv.util.Surv.from_arrays

See  
https://scikit-survival.readthedocs.io/en/latest/api/generated/sksurv.util.Surv.html#sksurv.util.Surv.from_arrays
